.github/workflows/docker-build.yaml

```diff
@@ -0,0 +1,31 @@
+on:
```

Comment:
I have a suggestion for this:

```
on:
  push:
  pull_request:


jobs:
  diff:
  # <...> job to perform a diff
  docker-image-build:
    strategy:
      matrix:
        apps: ${{ fromJSON(needs.diff.outputs.diff) }}
    # <...> Build docker image using your workflow, using matrix.apps, looping through it building and pushing
```

This way, we don't need to increase the paths in here https://github.com/conquer-project/votingapp/pull/2/files#diff-a894120de2d841cb7e1e40a5faee2225d8810cc3fe03ecd231b07322594fca1cR6 to each new folder.

Also it will generate one job for each app.

Reply:
I did not tested my suggestion, but I guess it work. It need to be adjusted as well.

References

- https://github.com/orgs/community/discussions/38088
- https://docs.github.com/en/actions/using-jobs/using-a-matrix-for-your-jobs#using-a-matrix-strategy
- https://stackoverflow.com/questions/65944700/how-to-run-git-diff-in-github-actions

Also for the git diff you do not have to use an action. It is just

`git diff --name-only HEAD~1` for main
`git diff --name-only HEAD ${{ github.ref_name }}` for non main